apps/system/js/activities.js

```diff
@@ -49,10 +104,26 @@
     * @param {Object} detail The activity choose event detail.
     */
     chooseActivity: function(detail) {
-      this._id = detail.id;
-      var choices = detail.choices;
+      var name = detail && detail.name;
+      var type = detail && detail.activityType;
+      var defaultConfig = DefaultActivityHelper.getDefaultConfig(name, type);
+      this._detail = detail;
       this.publish('activityrequesting');
-      if (choices.length === 1) {
+
+      if (defaultConfig) {
+        DefaultActivityHelper.getDefaultAction(name, type,
+          this._gotDefaultAction.bind(this));
```

Comment:
optional (because I should've suggested it during the first review) but easy fix: getDefaultAction should probably return a promise. it'll make this part more readable.